controlers/usersCtrl.js

```diff
@@ -0,0 +1,97 @@
+const User = require('../models/users.js');
+const Joi = require('joi');
+const jwt = require('jsonwebtoken');
+
+const userJoiSchema = Joi.object({
+    password: Joi.string().min(8).max(20).required(),
+    email: Joi.string().email().required(),
+    subscription: Joi.string().valid('starter', 'pro', 'business').default('starter'),
+    token: Joi.string().allow(null)
+})
+
+const createUser = async (req, res, next) => {
+    const { email, password } = req.body;
+    const user = await User.findOne({email}).lean();
+    const validateResult = userJoiSchema.validate(req.body);
+    if (validateResult.error) {
+        return res.status(400).json({message: 'Błąd z Joi lub innej biblioteki walidacji'})
+    }
+    if (user) {
+        return res.status(409).json({message: 'Email in use'});
+    }
+    try {
+        const newUser = new User({email, password});
+        await newUser.setPassword(password);
+        await newUser.save();
+        return res.status(201).json({message: `User ${req.body.email} created. Subscription: starter`});
+    } catch (err) {
+        next(err)
+    }
+};
+
+const loginUser = async (req, res) => {
+    const { email, password } = req.body;
+    const user = await User.findOne({email});
+    const validateResult = userJoiSchema.validate(req.body);
```

Comment:
Tutaj to samo, najpierw sprawdź dane a potem zapytanie do bazy danych. W obecnej sytuacji zakładasz że email jest poprawny, ale nie wiesz co fe ci wysle.

Reply:
Zrobione